### PR TITLE
update: cosign v1.2.1 to v1.3.0

### PR DESCRIPTION
**Pull request by xopham**
Fixes no issue

## Description

- updates cosign to v1.3.0
- fixes 1 vulnerabilities introduced via cosign
- some configuration and outputs of cosign were changed and had to be adjusted

## Checklist

- [x] PR is rebased to/aimed at branch `develop`
- [x] PR follows [Contributing Guide](../docs/CONTRIBUTING.md)
- [x] Added tests (if necessary)
- [x] Extended README/Documentation (if necessary)
- [x] Adjusted versions of image and Helm chart in `values.yaml` and `Chart.yaml` (if necessary)

### CODE REVIEW

File: connaisseur/validators/cosign/cosign_validator.py [hunk 1]
```diff
@@ -82,18 +82,14 @@ def __get_cosign_validated_digests(self, image: str, pubkey: str):
...
                 message=msg,
                 trust_data_type="dev.cosignproject.cosign/signature",
                 stderr=stderr,
             )
-        elif re.match(
-            r"^error: fetching signatures: getting signature manifest: "
-            r"GET https://[^ ]+ MANIFEST_UNKNOWN:.*",
-            stderr,
-        ):
+        elif "Error: no matching signatures:\n\nmain.go:48" in stderr:
```
Comment (Starkteetje) [inline connaisseur/validators/cosign/cosign_validator.py:92]: Using the line of error thrown seems very brittle. Is this intentional to find changes or can the line be removed to cause less work for us?

Comment (xopham) [reply connaisseur/validators/cosign/cosign_validator.py:92]: I would not see how we could distinguish between the two cases of no signature and wrong signature. Technically, would not need to have the line. Maybe I remove the exact line. That good for you?

Comment (xopham) [reply connaisseur/validators/cosign/cosign_validator.py]: done!

Comment (Starkteetje) [reply connaisseur/validators/cosign/cosign_validator.py]: Yeah, that works. Didn't really grasp that this was for distinguishing. Then main is actually needed if they changed the above error method. Feels kinda shitty, but I guess it's AGAIG unless there's different exit codes for cosign

**Review by Starkteetje — COMMENTED** [review]
lgtm

**Review by Starkteetje — APPROVED** [review]

**Pull request merged**